OIPS/oip-draft-versioning.md

```diff
@@ -0,0 +1,41 @@
+---
+oip: 4
+title: Releases Versioning Specification
+author:  Francesco Pacella <f@ost.com>
+discussions-to: -
+status: Draft
+type: Meta
+category -
+created: 2018-11-23
+requires : -
+replaces : -
+---
+
+#Releases Versioning Specification
+
+##Simple Summary
+We need to establish a new consistent versioning standard for OpenST releases going forward and agree on the versioning logic for Mosaic and BrandedToken and other repositories releases as well.
+
+##Abstract
+We started in November 2017 with an initial unique release stream, being OpenST. The first version was released under the name of 0.9.0. The versions that followed did not followed the common standard for releases, utilizing PATCH level to actually address MINOR level releases. 
+Referring here to standards as per https://semver.org/.
+The unique stream of releases has later split into three identifiable streams, which are independent from a release perspective. As the initial releases of the newly identified streams are imminent, the proper standard for release versioning shall be addressed at this point. 
+
+##Motivation
+We want to be clear, consistent and aligned with standards to facilitate the understanding of our development progresses for everyone.
+
+##Specification
```

Comment:
Please also specify the following:
- pre-releases should be additionally identified by "alpha", "rc" (for release candidates), etc. as agreed
- additional identifiers should be cardinally-numbered (e.g., starting with 1), such as "alpha.1", "rc.1"
- a [release candidate](https://en.wikipedia.org/wiki/Software_release_life_cycle#Release_candidate) "is a beta version with potential to be a final product", so should follow, not precede, an alpha
- versions are not prefixed with "v"
- format looks like: `<major>.<minor>.<patch>-<identifier>.<number>`, e.g. (`0.10.0-alpha.1`, `0.10.0-rc.1`)

The foregoing is specified in the semver documentation and thus does not necessarily need to spelled-out here. However, as some of these are small formatting distinctions that also depart from past practice (see, e.g., https://github.com/OpenSTFoundation/mosaic-contracts/releases/tag/v0.9.3-rc1), I believe there is value to explicit specification in this OIP.